client/components/spinner/index.jsx

```diff
@@ -30,30 +30,33 @@ const isSVGCSSAnimationSupported = ( () => {
 export default class Spinner extends PureComponent {
 	static propTypes = {
 		className: React.PropTypes.string,
+		delay: React.PropTypes.number,
 		size: React.PropTypes.number,
-		duration: React.PropTypes.number
+		counterClockwise: React.PropTypes.bool,
 	};
 
 	static instances = 0;
 
 	static defaultProps = {
+		delay: 100,
 		size: 20,
-		duration: 3000
+		counterClockwise: false,
 	};
 
-	constructor() {
-		super();
+	constructor( props ) {
+		super( props );
 		this.state = {
 			// We won't always have access to user-agent in server-side context, so
 			// initialize the spinner with fallback animations and check for support
 			// in componentDidMount()
-			isSVGCSSAnimationSupported: false
+			isSVGCSSAnimationSupported: false,
+			show: ! props.delay
```

Comment:
Currently, this will always equal to `false` initially because if the `delay` prop is not specified, it gets assigned value of `100` which makes this expression `false`.